src/hotspot/cpu/aarch64/pauth_aarch64.hpp

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright (c) 2021, Arm Limited. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ *
+ */
+
+#ifndef CPU_AARCH64_PAUTH_AARCH64_INLINE_HPP
+#define CPU_AARCH64_PAUTH_AARCH64_INLINE_HPP
+
+#include OS_CPU_HEADER_INLINE(pauth)
+
+inline bool pauth_ptr_is_raw(address ptr) {
+  // Confirm none of the high bits are set in the pointer.
```

Comment:
This predicate seems to me to be be misnamed: it's checking for unsigned/stripped, not for raw. The raw value is whatever gets saved in LR/pushed onto the stack.

Reply:
Ah, yes, because the name pauth_ptr_is_raw() now clashes with sender_pc_raw().

I'll fix up, one way or the other. Don't really want to call it pauth_ptr_is_unsigned() or pauth_ptr_is_authenticated(), because they seem to imply different things at first glance.

Reply:
raw is generally accepted as the correct name for a pointer that has not been signed (or has been authenticated/stripped). 
For example see: https://github.com/apple/llvm-project/blob/a63a81bd9911f87a0b5dcd5bdd7ccdda7124af87/clang/docs/PointerAuthentication.rst#basic-concepts

So, pauth_ptr_is_raw() is the correct name.

But, now what to rename sender_pc_raw() to ?  :)
The PAC coder would say:
   inline address  frame::sender_pc()       const { return *sender_pc_addr(); }
   inline address  frame::sender_pc_authenticated()           const { return pauth_strip_pointer(sender_pc()); }
But that's not right here for a common interface.

How about:
   inline address  frame::stored_sender_pc()       const { return *sender_pc_addr(); }
   inline address  frame::sender_pc()           const { return pauth_strip_pointer(stored_sender_pc()); }

Reply:
Off the top of my head, I would have thought that it makes the most sense to use Arm's terminology, which is "signed pointer". So, that'd be `frame::signed_sender_pc()` or `frame::maybe_signed_sender_pc()`.